src/interfaces/records/handlers/pruned-initial-records-write.ts

```diff
@@ -0,0 +1,32 @@
+import type { BaseMessage } from '../../../core/types.js';
+import type { EventLog } from '../../../event-log/event-log.js';
+import type { Readable } from 'readable-stream';
+import type { DataStore, MessageStore } from '../../../index.js';
+
+import { Message } from '../../../core/message.js';
+import { RecordsWriteHandler } from './records-write.js';
+
+/**
+ * Handler for initial `RecordsWrite` that already have its data pruned thus cannot supply data stream for it.
+ * NOTE: This is intended to be ONLY used by sync.
+ */
+export class PrunedInitialRecordsWriteHandler extends RecordsWriteHandler {
+  /**
+   * Stores the given message without storing the associated data.
+   * Requires `dataCid` to exist in the `RecordsWrite` message given.
+   */
+  public async storeMessage(
```

Comment:
I agree with you that this isn't my favorite approach. Inheritance always puts me edge. I don't want to block this PR, so I'll percolate on other ideas for it post-Miami.

Reply:
Looking forward to it!